base/ca/src/com/netscape/cms/servlet/cert/scep/CRSEnrollment.java

```diff
@@ -180,6 +181,9 @@
     private SecureRandom mRandom = null;
     private int mNonceSizeLimit = 0;
     private ICertificateAuthority ca;
+    private boolean mIsDynamicProfileId = false;
+    private String mAllowedDynamicProfileIdList = null;
+    private String[] mAllowedDynamicProfileId;
```

Comment:
Instead of `String[]` we could use `Collection<String>` so the check can be done as follows:
```
if (!mAllowedDynamicProfileId.contains(mProfileId)) {
```

Reply:
i.e. no need to create `isDynamicProfileIdAllowed()`.

Reply:
ok.

Reply:
is there an efficient way to do trim() on each element of the Collection?  I'll leave it as is for now.  Thanks!

Reply:
Supposedly it can be done like this:
```
collection.replaceAll(String::trim);
```
but I haven't tried that myself. Yeah, it's just a suggestion for possible enhancements. No need to implement in this PR.